pallets/tx-pause/src/lib.rs

```diff
@@ -135,21 +142,200 @@ pub mod pallet {
+                Self::deposit_event(Event::PalletPaused(pallet_name));
+            }
+
+            Ok(Some(T::WeightInfo::pause_transaction().saturating_mul(sum as Weight)).into())
+        }
+
+        /// Unpause all the calls of the listed pallets in `pallet_names`.
+        /// This logic is in its own extrinsic in order to not have to pause calls 1 by 1.
+        #[pallet::call_index(5)]
+        #[pallet::weight({
+            let len = pallet_names.len();
+            let max = T::MaxCallNames::get();
+            T::WeightInfo::pause_transaction().saturating_mul(len as Weight).saturating_mul(max as Weight)
+        })]
+        #[transactional]
+        pub fn unpause_pallets(
+            origin: OriginFor<T>,
+            pallet_names: Vec<Vec<u8>>,
+        ) -> DispatchResultWithPostInfo {
+            T::UnpauseOrigin::ensure_origin(origin)?;
+            let mut sum = 0;
+
+            for pallet_name in pallet_names {
+                let pallet_name_string = sp_std::str::from_utf8(&pallet_name)
+                    .map_err(|_| Error::<T>::InvalidCharacter)?;
+
+                let function_name =
+                    <CallOf<T> as GetCallMetadata>::get_call_names(pallet_name_string);
+                for call_name in function_name.to_vec() {
+                    let call_name = call_name.as_bytes().to_vec();
+
+                    PausedTransactions::<T>::take((&pallet_name, call_name));
+
+                    sum += 1;
+                }
+
+                // deposit event for each pallet
+                Self::deposit_event(Event::PalletUnpaused(pallet_name));
+            }
+
+            Ok(Some(T::WeightInfo::pause_transaction().saturating_mul(sum as Weight)).into())
+        }
+    }
+}
+
+impl<T: Config> Pallet<T> {
+    fn ensure_can_pause(pallet_name: &Vec<u8>) -> DispatchResult {
+        let pallet_name_string =
+            sp_std::str::from_utf8(pallet_name).map_err(|_| Error::<T>::InvalidCharacter)?;
+
+        // not allowed to pause calls of this pallet to ensure safe
+        ensure!(
+            pallet_name_string != <Self as PalletInfoAccess>::name(),
+            Error::<T>::CannotPause
+        );
+
+        // not allowed to pause `UnpausablePallets`
+        ensure!(
+            !T::UnpausablePallets::contains(pallet_name),
+            Error::<T>::CannotPause
+        );
+
+        Ok(())
+    }
+
+    fn pause_one(
+        pallet_name: &Vec<u8>,
+        function_name: &Vec<u8>,
+        deposit_event: bool,
+    ) -> DispatchResult {
+        PausedTransactions::<T>::mutate_exists((pallet_name, function_name), |maybe_paused| {
+            if maybe_paused.is_none() {
+                *maybe_paused = Some(());
+                if deposit_event {
+                    Self::deposit_event(Event::TransactionPaused(
+                        pallet_name.clone(),
+                        function_name.clone(),
+                    ));
+                }
+            }
+        });
+        Ok(())
+    }
```

Comment:
It may be ok to just emit a bunch of events for simplicity in the event mapping. Its not a call that will be used regularly so the overhead incurred isn't particularly costly.

This way is fine as is too, I don't feel strongly